src/test/java/uk/ac/ebi/ampt2d/accessioning/commons/persistence/BasicSpringDataRepositoryDatabaseServiceTest.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.test.context.transaction.TestTransaction;
+import uk.ac.ebi.ampt2d.accessioning.commons.generators.ModelHashAccession;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.persistence.TestEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestRepository;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestDatabaseServiceTestConfiguration.class})
+public class BasicSpringDataRepositoryDatabaseServiceTest {
+
+    @Autowired
+    private TestRepository repository;
+
+    @Autowired
+    private BasicSpringDataRepositoryDatabaseService<TestModel, TestEntity, String, String> service;
+
+    @Test
+    public void testFindInEmptyRepository() {
+        assertEquals(0, service.findAllAccessionByAccessions(Arrays.asList("a1", "a2")).size());
+        assertEquals(0, service.findAllAccessionsByHash(Arrays.asList("h1", "h2")).size());
+        assertEquals(0, service.getExistingAccessions(Arrays.asList("h1", "h2")).size());
+    }
+
+    @Test
+    public void saveUniqueElements() {
+        service.save(Arrays.asList(
+                ModelHashAccession.of(TestModel.of("something1"), "h1", "a1"),
+                ModelHashAccession.of(TestModel.of("something2"), "h2", "a2"),
+                ModelHashAccession.of(TestModel.of("something3"), "h3", "a3")
+        ));
+
+        Map<String, ? extends TestModel> accessionsToModels = service.findAllAccessionByAccessions(
+                Arrays.asList("a1", "a2"));
+        assertEquals(2, accessionsToModels.size());
+        assertEquals("something1", accessionsToModels.get("a1").getSomething());
+        assertEquals("something2", accessionsToModels.get("a2").getSomething());
+
+        Map<String, ? extends TestModel> accessionsToModels2 = service.findAllAccessionsByHash(
+                Arrays.asList("h1", "h2"));
+        assertEquals(2, accessionsToModels2.size());
+        assertEquals("something1", accessionsToModels2.get("a1").getSomething());
+        assertEquals("something2", accessionsToModels2.get("a2").getSomething());
+
+        Map<String, String> hashToAccession = service.getExistingAccessions(Arrays.asList("h1", "h3"));
+        assertEquals(2, hashToAccession.size());
+        assertEquals("a1", hashToAccession.get("h1"));
+        assertEquals("a3", hashToAccession.get("h3"));
+    }
+
+    @Test(expected = DataIntegrityViolationException.class)
+    @Commit
+    public void saveNonUniqueElements() {
+        service.save(Arrays.asList(
+                ModelHashAccession.of(TestModel.of("something1"), "h1", "a1"),
+                ModelHashAccession.of(TestModel.of("something2"), "h1", "a2"),
+                ModelHashAccession.of(TestModel.of("something3"), "h3", "a3")
+        ));
+        TestTransaction.end();
```

Comment:
Is it necessary to provide both the annotation and the method call? The Javadoc doesn't make it very clear.

Reply:
The annotation will flag the transaction to commit it, otherwise it will be rolledback when end is called, end is called at that point to make the test, otherwise, the error will slip and will be handler by the transaction handler of Spring test and we won't receive the error.